### PR TITLE
feat: add prepareBarChartData function in chart package

### PR DESCRIPTION
**Pull request by mikasackermn**
# Summary

 add prepareBarChartData function in chart package
 
Fixes # (issue)

# How did you test this change?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Implemented a user interface (UI) change, referencing our Figma design to ensure pixel-perfect precision.

### CODE REVIEW

**Comment by nikaaru** [comment]
Please make the naming general so that it is meaningful for use anywhere, I suggest that wherever **`blockchain`** is used, you can use a **`bucket`** instead.

File: widget/charts/src/components/BarChart/BarChart.types.ts [hunk 1]
```diff
@@ -30,3 +30,27 @@ export type TooltipDataType = {
...
+export type DailyDataType = {
+  date: string;
+  bucket: string;
+  count: number;
+  volume: number;
+  txCountEffective: number;
+  volumeEffective: number;
+};
+
+export type ChartOptionsType =
```
Comment (nikaaru) [inline widget/charts/src/components/BarChart/BarChart.types.ts:43]: Please remove the **`isStackBar`**, add a **`defaultLabel`** to it, and handle it without a stack inside this method.
Also, we can pass the color as a single-member array in the non-stack mode

Comment (mikasackermn) [reply widget/charts/src/components/BarChart/BarChart.types.ts]: fixed

File: widget/charts/src/components/BarChart/BarChart.constants.ts [hunk 2]
```diff
@@ -31,3 +30,5 @@ export const bottomAxisData = {
     90: { numBottomAxis: 3, startBottomAxis: 10, intervalBottomAxis: 30 },
   },
 };
+
+export const BAR_CHART_BUCKET_NUMBER = 10;
```
Comment (nikaaru) [inline widget/charts/src/components/BarChart/BarChart.constants.ts:34]:
```suggestion
export const MAX_BAR_BUCKETS = 10;
```

File: widget/charts/src/components/BarChart/BarChart.helpers.ts [hunk 3]
```diff
@@ -54,3 +61,87 @@ export const getDaysRange = (lengthValue: number) => {
...
+  // create map structure for assign color for each bucket
+  topBucket.forEach((bucketItem, index) => {
+    colorBucketMap.set(
+      bucketItem,
+      barChartColors[index % barChartColors.length]
+    );
+    buckets.push(bucketItem);
+  });
+
+  if (buckets.length === BAR_CHART_BUCKET_NUMBER) {
```
Comment (nikaaru) [inline widget/charts/src/components/BarChart/BarChart.helpers.ts:100]: please use size of map:
```
  const bucketCount = sumBucketMap.size;
  if (bucketCount > MAX_BAR_BUCKETS)
```

File: widget/charts/src/components/BarChart/BarChart.helpers.ts [hunk 4]
```diff
@@ -54,3 +61,87 @@ export const getDaysRange = (lengthValue: number) => {
...
+  const sumBucketMap = new Map<string, number>();
+  dailyData.forEach((dailyItem) => {
+    const keyItem = dailyItem.bucket || label;
+
+    const sum = sumBucketMap.get(keyItem) || 0;
+    const newValue = dailyItem.value;
+    sumBucketMap.set(keyItem, sum + newValue);
+  });
+
+  // sort buckets with sum value base on type (transaction or volume)
```
Comment (nikaaru) [inline widget/charts/src/components/BarChart/BarChart.helpers.ts:83]: we can remove this comment

File: widget/charts/src/components/BarChart/BarChart.helpers.ts [hunk 5]
```diff
@@ -54,3 +61,87 @@ export const getDaysRange = (lengthValue: number) => {
...
+
+export const prepareBarChartData = (chartOption: ChartOptionsType) => {
+  const { dailyData, label = 'Count' } = chartOption;
+  const chartData: BarStackDataType[] = [];
+  const colorBucketMap: ColorBucketMapType = new Map();
+  const buckets: string[] = [];
+
+  const { barChartColors } = chartOption;
+
+  // map sum of value base on type(transaction or volume) for each bucket
```
Comment (nikaaru) [inline widget/charts/src/components/BarChart/BarChart.helpers.ts:73]:
```suggestion
  // map sum of value for each bucket
```

**Comment by nikaaru** [comment]
NIT: 
suggestion commit message:  fix: add prepare data function for chart package

File: widget/charts/src/components/BarChart/BarChart.constants.ts (hunk 2 above)
Comment (mikasackermn) [reply widget/charts/src/components/BarChart/BarChart.constants.ts:34]: Fixed

File: widget/charts/src/components/BarChart/BarChart.helpers.ts (hunk 3 above)
Comment (mikasackermn) [reply widget/charts/src/components/BarChart/BarChart.helpers.ts:100]: Fixed

File: widget/charts/src/components/BarChart/BarChart.helpers.ts (hunk 4 above)
Comment (mikasackermn) [reply widget/charts/src/components/BarChart/BarChart.helpers.ts:83]: Fixed

File: widget/charts/src/components/BarChart/BarChart.helpers.ts (hunk 5 above)
Comment (mikasackermn) [reply widget/charts/src/components/BarChart/BarChart.helpers.ts:73]: Fixed

File: widget/charts/src/components/BarChart/BarChart.helpers.ts [hunk 6]
```diff
@@ -54,3 +61,87 @@ export const getDaysRange = (lengthValue: number) => {
...
+          : '0';
+      });
+
+    topBucket.forEach((topItem) => {
+      if (!(topItem in dataItem)) {
+        dataItem[topItem] = '0';
+      }
+    });
+
+    if (buckets.length > BAR_CHART_BUCKET_NUMBER) {
```
[the remark this answers is not in the document]
Comment (mikasackermn) [reply widget/charts/src/components/BarChart/BarChart.helpers.ts:134]: Fixed

**Review by nikaaru — APPROVED** [review]
LGTM

**Pull request merged**